gsgmail_process_email.py

```diff
@@ -1094,4 +1107,18 @@ def _create_dict_hash(self, input_dict):
             self._base_connector.debug_print('Exception: ', e)
             return None
 
-        return hashlib.md5(input_dict_str.encode('utf-8')).hexdigest()
+        fips_enabled = self._get_fips_enabled()
+        # if fips is not enabled, we should continue with our existing md5 usage for generating hashes
+        # to not impact existing customers
+        try:
+            dict_hash = input_dict_str.encode('utf-8')
+            if not fips_enabled:
+                dict_hash = hashlib.md5(dict_hash)
+            else:
+                dict_hash = hashlib.sha256(dict_hash)
+        except:
+            if not fips_enabled:
+                dict_hash = hashlib.md5(dict_hash.encode())
+            else:
+                dict_hash = hashlib.sha256(dict_hash.encode())
+        return dict_hash.hexdigest()
```

Comment:
`hexdigest` should also be callable on sha256 (see [example](https://github.com/splunk-soar-connectors/alexa/blob/159d160116f32fae779e2191ae648986cce06027/alexa_connector.py#L126))